src/js/node/http2.ts

```diff
@@ -7,27 +7,963 @@ const { hideFromStack, throwNotImplemented } = require("internal/shared");
+const bunSocketServerOptions = Symbol.for("::bunnetserveroptions::");
+const bunSocketInternal = Symbol.for("::bunnetsocketinternal::");
+const kInfoHeaders = Symbol("sent-info-headers");
+
+const Stream = require("node:stream");
+const { Readable } = Stream;
+type Http2ConnectOptions = {
+  settings?: Settings;
+  protocol?: "https:" | "http:";
+  createConnection?: Function;
+};
 const TLSSocket = tls.TLSSocket;
+const Socket = net.Socket;
 const EventEmitter = require("node:events");
 const { Duplex } = require("node:stream");
-const primordials = require("internal/primordials");
 
-const [H2FrameParser, getPackedSettings, getUnpackedSettings] = $zig("h2_frame_parser.zig", "createNodeHttp2Binding");
+const {
+  FunctionPrototypeBind,
+  StringPrototypeTrim,
+  ArrayPrototypePush,
+  ObjectAssign,
+  ArrayIsArray,
+  SafeArrayIterator,
+  StringPrototypeToLowerCase,
+  StringPrototypeIncludes,
+  ObjectKeys,
+  ObjectPrototypeHasOwnProperty,
+  SafeSet,
+  DatePrototypeToUTCString,
+  DatePrototypeGetMilliseconds,
+} = require("internal/primordials");
+const RegExpPrototypeExec = RegExp.prototype.exec;
+
+const [H2FrameParser, assertSettings, getPackedSettings, getUnpackedSettings] = $zig(
+  "h2_frame_parser.zig",
+  "createNodeHttp2Binding",
+);
 
 const sensitiveHeaders = Symbol.for("nodejs.http2.sensitiveHeaders");
 const bunHTTP2Native = Symbol.for("::bunhttp2native::");
-const bunHTTP2StreamResponded = Symbol.for("::bunhttp2hasResponded::");
 const bunHTTP2StreamReadQueue = Symbol.for("::bunhttp2ReadQueue::");
-const bunHTTP2Closed = Symbol.for("::bunhttp2closed::");
+
 const bunHTTP2Socket = Symbol.for("::bunhttp2socket::");
-const bunHTTP2WantTrailers = Symbol.for("::bunhttp2WantTrailers::");
+const bunHTTP2StreamFinal = Symbol.for("::bunHTTP2StreamFinal::");
+
+const bunHTTP2StreamStatus = Symbol.for("::bunhttp2StreamStatus::");
+
 const bunHTTP2Session = Symbol.for("::bunhttp2session::");
+const bunHTTP2Headers = Symbol.for("::bunhttp2headers::");
 
 const ReflectGetPrototypeOf = Reflect.getPrototypeOf;
-const FunctionPrototypeBind = primordials.FunctionPrototypeBind;
-const StringPrototypeSlice = String.prototype.slice;
+
+const kBeginSend = Symbol("begin-send");
+const kServer = Symbol("server");
+const kState = Symbol("state");
+const kStream = Symbol("stream");
+const kResponse = Symbol("response");
+const kHeaders = Symbol("headers");
+const kRawHeaders = Symbol("rawHeaders");
+const kTrailers = Symbol("trailers");
+const kRawTrailers = Symbol("rawTrailers");
+const kSetHeader = Symbol("setHeader");
+const kAppendHeader = Symbol("appendHeader");
+const kAborted = Symbol("aborted");
+const kRequest = Symbol("request");
+
+function validateString(value, name) {
+  if (typeof value !== "string") throw ERR_INVALID_ARG_TYPE(name, "string", value);
+}
+hideFromStack(validateString);
+
+function validateFunction(value, name) {
```

Comment:
Please reuse the existing validators for these functions instead of adding another implementation.